internal/cluster/clusteradapter/client_factory.go

```diff
@@ -27,6 +28,39 @@ import (
 	"github.com/banzaicloud/pipeline/pkg/k8sclient"
 )
 
+// ClientFactory returns a Kubernetes client.
+type ClientFactory struct {
+	secretStore common.SecretStore
+}
+
+// NewClientFactory returns a new ClientFactory.
+func NewClientFactory(secretStore common.SecretStore) ClientFactory {
+	return ClientFactory{
+		secretStore: secretStore,
+	}
+}
+
+// FromSecret creates a Kubernetes client for a cluster from a secret.
+func (f ClientFactory) FromSecret(ctx context.Context, secretID string) (kubernetes.Interface, error) {
+	values, err := f.secretStore.GetSecretValues(ctx, secretID)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: better secret parsing?
```

Comment:
Would be nice to have a common parser for this (not related to this PR), I think we have this code snippet at least 8 different places.

Reply:
Agreed 🙂

Reply:
My idea was to have secret specific access interfaces which expose a secret in a denormalized form.

Reply:
Great!